src/modem/mod.rs

```diff
@@ -268,7 +280,27 @@ impl<'c, P: ModemPower> Modem<'c, P> {
                 .await
         )?;
 
-        self.wait_for_registration().await?;
+        if self.automatic_registration {
+            let active_mode = self.automatic_registration(&commands).await?;
+            match active_mode {
+                SystemMode::Gsm => {
+                    self.network_priority =
+                        [SystemMode::Gsm, SystemMode::LteCatM1, SystemMode::LteNbIot]
+                }
+                SystemMode::LteCatM1 => {
+                    self.network_priority =
+                        [SystemMode::LteCatM1, SystemMode::Gsm, SystemMode::LteNbIot]
+                }
+                SystemMode::LteNbIot => {
+                    // We don't really want NbIoT, let's keep the previous priority
+                }
```

Comment:
This is an open source lib, we shouldn't make assumptions based on our own use-case like this. Other people might not want to de-prioritise NbIot